lib/temp-directory.js

```diff
@@ -11,10 +11,20 @@ function create(context, next) {
   var _path = path.join(osenv.tmpdir(),uuid.v4());
   var npmConfigTmp = path.join(_path, 'npm_config_tmp');
 
-  context.path = _path;
+  if (context.options && context.options.tmpDir) {
+    context.path = path.join(context.options.tmpDir, context.module.name);
+    context.emit('data', 'verbose', context.module.name + ' mk.tempdir', context.path);
+    mkdirp(context.path, function (err) {
```

Comment:
this async call is not handled correctly. I could potentially end up calling the callback twice.